packages/runtime/datastore/src/remoteChannelContext.ts

```diff
@@ -191,62 +189,23 @@ export class RemoteChannelContext implements IChannelContext {
 				attributesBlobKey,
 			);
 		}
-
-		let factory: IChannelFactory | undefined;
-		// this is a backward compatibility case where
-		// the attach message doesn't include
-		// the attributes. Since old attach messages
-		// will not have attributes we need to keep
-		// this as long as we support old attach messages
-		if (attributes === undefined) {
-			if (this.attachMessageType === undefined) {
-				// TODO: dataStoreId may require a different tag from PackageData #7488
-				throw new DataCorruptionError("channelTypeNotAvailable", {
-					channelId: {
-						value: this.id,
-						tag: TelemetryDataTag.CodeArtifact,
-					},
-					dataStoreId: {
-						value: this.dataStoreContext.id,
-						tag: TelemetryDataTag.CodeArtifact,
-					},
-					dataStorePackagePath: this.dataStoreContext.packagePath.join("/"),
-				});
-			}
-			factory = this.registry.get(this.attachMessageType);
-			if (factory === undefined) {
-				// TODO: dataStoreId may require a different tag from PackageData #7488
-				throw new DataCorruptionError("channelFactoryNotRegisteredForAttachMessageType", {
-					channelId: {
-						value: this.id,
-						tag: TelemetryDataTag.CodeArtifact,
-					},
-					dataStoreId: {
-						value: this.dataStoreContext.id,
-						tag: TelemetryDataTag.CodeArtifact,
-					},
-					dataStorePackagePath: this.dataStoreContext.packagePath.join("/"),
-					channelFactoryType: this.attachMessageType,
-				});
-			}
-			attributes = factory.attributes;
-		} else {
-			factory = this.registry.get(attributes.type);
-			if (factory === undefined) {
-				// TODO: dataStoreId may require a different tag from PackageData #7488
-				throw new DataCorruptionError("channelFactoryNotRegisteredForGivenType", {
-					channelId: {
-						value: this.id,
-						tag: TelemetryDataTag.CodeArtifact,
-					},
-					dataStoreId: {
-						value: this.dataStoreContext.id,
-						tag: TelemetryDataTag.CodeArtifact,
-					},
-					dataStorePackagePath: this.dataStoreContext.packagePath.join("/"),
-					channelFactoryType: attributes.type,
-				});
-			}
+		assert(attributes !== undefined, "Channel attributes blob missing");
```

Comment:
If I'm reading it correctly, then we could have really old file at rest that will be loaded in the future and will get here with attributes === undefined. And if so, it feels like it will stop working. Chances are low, but we do have a promise that all old file formats will be supported for reading.
If I got it right, maybe first add telemetry that it does not happen at all, and then remove it? Likely it will point out that such cases do not happen, so we are fine. Not sure.

Reply:
We already have telemetry. The data corruption errors are not hit today. But we only have 30 days of data.

Reply:
i wouldn't necessarily remove the backcompat, but collapse all the checks to a single assert, basically we don't expect malformed old data, but we accept well formed old data.

Reply:
i think you'll need to backcompat to pass the snapshot tests anyway.

Reply:
Updated it to collapse the checks. Can you please take a look now.